tests/components/neato/test_config_flow.py

```diff
@@ -1,160 +1,156 @@
-    flow = init_config_flow(hass)
+async def test_reauth(hass: HomeAssistantType):
+    """Test initialization of the reauth flow."""
+    MockConfigEntry(
+        domain=NEATO_DOMAIN,
+        data={"username": "abcdef", "password": "123456", "vendor": "neato"},
+        version=1,
+    ).add_to_hass(hass)
 
-    result = await flow.async_step_import(
-        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD, CONF_VENDOR: VENDOR_NEATO}
+    # Should show form
+    result = await hass.config_entries.flow.async_init(
+        "neato", context={"source": config_entries.SOURCE_REAUTH}
     )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "reauth_confirm"
 
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == f"{USERNAME} (from configuration)"
-    assert result["data"][CONF_USERNAME] == USERNAME
-    assert result["data"][CONF_PASSWORD] == PASSWORD
-    assert result["data"][CONF_VENDOR] == VENDOR_NEATO
+    # Form confirmed, init new flow
+    with patch(
+        "homeassistant.components.neato.config_flow.OAuth2FlowHandler.async_step_user",
+        return_value={
+            "description_placeholders": None,
+            "flow_id": result["flow_id"],
+            "handler": "neato",
+            "step_id": "auth",
+            "type": "external",
+            "url": "https://example.com/",
+        },
+    ) as mock_user:
+        await hass.config_entries.flow.async_configure(result["flow_id"])
 
+    mock_user.assert_called_once_with()
+
+
+async def test_oauth_create_entry(hass: HomeAssistantType, local_impl):  # noqa: 811
+    """Test migration from version 1 to version 2."""
+    flow = OAuth2FlowHandler()
+    flow.hass = hass
+    flow.flow_impl = local_impl
 
-async def test_abort_if_already_setup(hass, account):
-    """Test we abort if Neato is already setup."""
-    flow = init_config_flow(hass)
     MockConfigEntry(
+        entry_id="my_entry",
         domain=NEATO_DOMAIN,
-        data={
-            CONF_USERNAME: USERNAME,
-            CONF_PASSWORD: PASSWORD,
-            CONF_VENDOR: VENDOR_NEATO,
-        },
+        data={"username": "abcdef", "password": "123456", "vendor": "neato"},
+        version=1,
     ).add_to_hass(hass)
 
-    # Should fail, same USERNAME (import)
-    result = await flow.async_step_import(
-        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD, CONF_VENDOR: VENDOR_NEATO}
-    )
-    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
-    assert result["reason"] == "already_configured"
-
-    # Should fail, same USERNAME (flow)
-    result = await flow.async_step_user(
-        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD, CONF_VENDOR: VENDOR_NEATO}
-    )
-    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
-    assert result["reason"] == "already_configured"
+    # Update entry
+    with patch(
+        "homeassistant.config_entries.ConfigEntries.async_reload",
```

Comment:
It's better to patch the integration `async_setup_entry` and check number of calls to that.

Reply:
Done.

Reply:
We're using the config entries api in the tests, so we should avoid patching that.